master/v5.0.x: Compile error with clang and armclang

Thank you for taking the time to submit an issue!

## Background information

Building latest ompi out of gtihub, get error with armclang 21.1

### What version of Open MPI are you using? (e.g., v3.0.5, v4.0.2, git branch name and hash, etc.)

github @ 4b50f265e6cbe31d744af2a4e47a28d73ac5e4d0 (master)

### Describe how Open MPI was installed (e.g., from a source/distribution tarball, from a git clone, from an operating system distribution package, etc.)

clone

### If you are building/installing from a git clone, please copy-n-paste the output from `git submodule status`.

```
 409d89fd2c6ad47e6cab4d4f18e72dd3e0af2e70 3rd-party/openpmix (v1.1.3-3138-g409d89fd)
 8622c287399f3a22102b998aee098d97a621bdbe 3rd-party/prrte (psrvr-v2.0.0rc1-4018-g8622c28739)
```

### Please describe the system on which you are running

* Operating system/version: CentOS Linux release 8.1.1911
* Computer hardware: aarch64
* Network type: IB

-----------------------------

## Details of the problem

```
make[2]: Entering directory '/autofs/nccs-svm1_envoy_od/arcurtis/src/openmpi/build/ompi/tools/ompi_info'
depbase=`echo param.o | sed 's|[^/]*$|.deps/&|;s|\.o$||'`;\
armclang -DHAVE_CONFIG_H -I. -I../../../../openmpi-git/ompi/tools/ompi_info -I../../../opal/include -I../../../ompi/include -I../../../oshmem/include -I../../../ompi/mpiext/cuda/c   -iquote../../../../openmpi-git -iquote../../.. -iquote../../../../openmpi-git/opal/include -iquote../../../../openmpi-git/ompi/include -iquote../../../../openmpi-git/oshmem/include  -I/ccsopen/home/arcurtis/src/openmpi/build/3rd-party/libevent-2.1.12-stable -I/ccsopen/home/arcurtis/src/openmpi/build/3rd-party/libevent-2.1.12-stable/include -I/ccsopen/home/arcurtis/src/openmpi/build/3rd-party/hwloc-2.4.0/include -I/ccsopen/home/arcurtis/src/openmpi/openmpi-git/3rd-party/hwloc-2.4.0/include -I/ccsopen/home/arcurtis/src/openmpi/build/3rd-party/openpmix/include -I/ccsopen/home/arcurtis/src/openmpi/openmpi-git/3rd-party/openpmix/include -DOPAL_CONFIGURE_USER="\"arcurtis\"" -DOPAL_CONFIGURE_HOST="\"wombat-login2\"" -DOPAL_CONFIGURE_DATE="\"Thu Oct  7 19:26:30 UTC 2021\"" -DOMPI_BUILD_USER="\"$USER\"" -DOMPI_BUILD_HOST="\"${HOSTNAME:-`(hostname || uname -n) | sed 1q`}\"" -DOMPI_BUILD_DATE="\"`../../../../openmpi-git/config/getdate.sh`\"" -DOMPI_BUILD_CFLAGS="\"-O3 -DNDEBUG -finline-functions\"" -DOMPI_BUILD_CPPFLAGS="\"-iquote../../../../openmpi-git -iquote../../.. -iquote../../../../openmpi-git/opal/include -iquote../../../../openmpi-git/ompi/include -iquote../../../../openmpi-git/oshmem/include  -I/ccsopen/home/arcurtis/src/openmpi/build/3rd-party/libevent-2.1.12-stable -I/ccsopen/home/arcurtis/src/openmpi/build/3rd-party/libevent-2.1.12-stable/include -I/ccsopen/home/arcurtis/src/openmpi/build/3rd-party/hwloc-2.4.0/include -I/ccsopen/home/arcurtis/src/openmpi/openmpi-git/3rd-party/hwloc-2.4.0/include -I/ccsopen/home/arcurtis/src/openmpi/build/3rd-party/openpmix/include -I/ccsopen/home/arcurtis/src/openmpi/openmpi-git/3rd-party/openpmix/include\"" -DOMPI_BUILD_CXXFLAGS="\"-DNDEBUG \"" -DOMPI_BUILD_CXXCPPFLAGS="\"@CXXCPPFLAGS@\"" -DOMPI_BUILD_FFLAGS="\"\"" -DOMPI_BUILD_FCFLAGS="\"\"" -DOMPI_BUILD_LDFLAGS="\"\"" -DOMPI_BUILD_LIBS="\"-lpthread -lrt -lutil \"" -DOPAL_CC_ABSOLUTE="\"/sw/wombat/ARM_Compiler/20.3/arm-linux-compiler-20.3_Generic-AArch64_RHEL-8_aarch64-linux/bin/armclang\"" -DOMPI_CXX_ABSOLUTE="\"/sw/wombat/ARM_Compiler/20.3/arm-linux-compiler-20.3_Generic-AArch64_RHEL-8_aarch64-linux/bin/armclang++\"" -O3 -DNDEBUG -finline-functions -MT param.o -MD -MP -MF $depbase.Tpo -c -o param.o ../../../../openmpi-git/ompi/tools/ompi_info/param.c &&\
mv -f $depbase.Tpo $depbase.Po
../../../../openmpi-git/ompi/tools/ompi_info/param.c:341:80: error: expected ')'
                  PLATFORM_STRINGIFY(OPAL_BUILD_PLATFORM_COMPILER_VERSION_STR));
                                                                               ^
../../../../openmpi-git/ompi/tools/ompi_info/param.c:340:18: note: to match this '('
    opal_info_out("C compiler version", "compiler:c:version",
                 ^
1 error generated.
```

--- a/config/opal_check_compiler_version.m4
+++ b/config/opal_check_compiler_version.m4
@@ -22,9 +22,7 @@ dnl
 AC_DEFUN([OPAL_CHECK_COMPILER_VERSION_ID],
 [
     OPAL_CHECK_COMPILER(FAMILYID)
-    OPAL_CHECK_COMPILER_STRINGIFY(FAMILYNAME)
     OPAL_CHECK_COMPILER(VERSION)
-    OPAL_CHECK_COMPILER_STRING(VERSION_STR)
 ])dnl
 
 
@@ -51,67 +49,6 @@ AC_DEFUN([OPAL_CHECK_COMPILER], [
                 opal_cv_compiler_$1=0
             ], [
                 opal_cv_compiler_$1=0
-            ])
-            CPPFLAGS=$CPPFLAGS_orig
-    ])
-    AC_DEFINE_UNQUOTED([OPAL_BUILD_PLATFORM_COMPILER_$1], [$opal_cv_compiler_$1],
-                       [The compiler $lower which OMPI was built with])
-])dnl
-
-AC_DEFUN([OPAL_CHECK_COMPILER_STRING], [
-    AS_LITERAL_IF([$1], [],
-                  [m4_fatal([OPAL_CHECK_COMPILER_STRING argument must be a literal])])
-    lower=m4_tolower([$1])
-    AC_CACHE_CHECK([for compiler $lower], [opal_cv_compiler_$1],
-    [
-            CPPFLAGS_orig=$CPPFLAGS
-            CPPFLAGS="-I${OPAL_TOP_SRCDIR}/opal/include $CPPFLAGS"
-            AC_RUN_IFELSE([AC_LANG_PROGRAM([[
-#include <stdio.h>
-#include <stdlib.h>
-#include "opal/opal_portable_platform.h"
-]],[[
-    FILE * f;
-    f=fopen("conftestval", "w");
-    if (!f) exit(1);
-    fprintf (f, "%s", PLATFORM_COMPILER_$1);
-            ]])], [
-                opal_cv_compiler_$1=`cat conftestval`
-            ], [
-                opal_cv_compiler_$1=UNKNOWN
-            ], [
-                opal_cv_compiler_$1=UNKNOWN
-            ])
-            CPPFLAGS=$CPPFLAGS_orig
-    ])
-    AC_DEFINE_UNQUOTED([OPAL_BUILD_PLATFORM_COMPILER_$1], [$opal_cv_compiler_$1],
-                       [The compiler $lower which OMPI was built with])
-])dnl
-
-
-AC_DEFUN([OPAL_CHECK_COMPILER_STRINGIFY], [
-    AS_LITERAL_IF([$1], [],
-                  [m4_fatal([OPAL_CHECK_COMPILER_STRINGIFY argument must be a literal])])
-    lower=m4_tolower([$1])
-    AC_CACHE_CHECK([for compiler $lower], [opal_cv_compiler_$1],
-    [
-            CPPFLAGS_orig=$CPPFLAGS
-            CPPFLAGS="-I${OPAL_TOP_SRCDIR}/opal/include $CPPFLAGS"
-            AC_RUN_IFELSE([AC_LANG_PROGRAM([[
-#include <stdio.h>
-#include <stdlib.h>
-#include "opal/opal_portable_platform.h"
-]],[[
-    FILE * f;
-    f=fopen("conftestval", "w");
-    if (!f) exit(1);
-    fprintf (f, "%s", PLATFORM_STRINGIFY(PLATFORM_COMPILER_$1));
-            ]])], [
-                opal_cv_compiler_$1=`cat conftestval`
-            ], [
-                opal_cv_compiler_$1=UNKNOWN
-            ], [
-                opal_cv_compiler_$1=UNKNOWN
             ])
             CPPFLAGS=$CPPFLAGS_orig
     ])

--- a/ompi/tools/ompi_info/param.c
+++ b/ompi/tools/ompi_info/param.c
@@ -40,6 +40,7 @@
 
 #include MCA_timer_IMPLEMENTATION_HEADER
 #include "opal/include/opal/version.h"
+#include "opal/opal_portable_platform.h"
 #include "opal/class/opal_value_array.h"
 #include "opal/class/opal_pointer_array.h"
 #include "opal/util/printf.h"
@@ -47,7 +48,6 @@
 #include "opal/runtime/opal_info_support.h"
 
 #include "ompi/tools/ompi_info/ompi_info.h"
-#include "ompi/include/mpi_portable_platform.h"
 
 
 const char *ompi_info_deprecated_value = "deprecated-ompi-info-value";
@@ -336,9 +336,9 @@ void ompi_info_do_config(bool want_all)
     opal_info_out("C compiler absolute", "compiler:c:absolute",
                   OPAL_CC_ABSOLUTE);
     opal_info_out("C compiler family name", "compiler:c:familyname",
-                  PLATFORM_STRINGIFY(OPAL_BUILD_PLATFORM_COMPILER_FAMILYNAME));
+                  PLATFORM_STRINGIFY(PLATFORM_COMPILER_FAMILYNAME));
     opal_info_out("C compiler version", "compiler:c:version",
-                  PLATFORM_STRINGIFY(OPAL_BUILD_PLATFORM_COMPILER_VERSION_STR));
+                  PLATFORM_COMPILER_VERSION_STR);
 
     if (want_all) {
         opal_info_out_int("C char size", "compiler:c:sizeof:char", sizeof(char));

--- a/oshmem/tools/oshmem_info/param.c
+++ b/oshmem/tools/oshmem_info/param.c
@@ -147,9 +147,9 @@ void oshmem_info_do_config(bool want_all)
     opal_info_out("C compiler absolute", "compiler:c:absolute",
                   OPAL_CC_ABSOLUTE);
     opal_info_out("C compiler family name", "compiler:c:familyname",
-                  PLATFORM_STRINGIFY(OPAL_BUILD_PLATFORM_COMPILER_FAMILYNAME));
+                  PLATFORM_STRINGIFY(PLATFORM_COMPILER_FAMILYNAME));
     opal_info_out("C compiler version", "compiler:c:version",
-                  PLATFORM_STRINGIFY(OPAL_BUILD_PLATFORM_COMPILER_VERSION_STR));
+                  PLATFORM_COMPILER_VERSION_STR);
 
     if (want_all) {
         opal_info_out_int("C char size", "compiler:c:sizeof:char", sizeof(char));

